File: src/com/colacelli/irclib/connection/listeners/OnServerMessageListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.connection.Connection;

public interface OnServerMessageListener extends Listener {
    Type TYPE = Type.SERVER_MESSAGE;

    String serverMessage();

    void onServerMessage(Connection connection, String message, String command, String... args);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/main/kotlin/com/colacelli/irclib/connection/listeners/OnServerMessageListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.connection.Connection

interface OnServerMessageListener : Listener {
    fun serverMessage(): String
    fun onServerMessage(connection: Connection, message: String, command: String, args: List<String>)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.SERVER_MESSAGE
    }
}
